src/main/java/Duke.java

```diff
@@ -1,10 +1,36 @@
+import java.io.File;
+
+/**
+ * Class responsible for the execution of the Duke chatbot.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private final String DATA_PATH_NAME = "./data";
+    private final String DATA_FILE_PATH = "./data/dukedata.txt";
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initialises a new Duke object using the specified DATA_PATH_NAME and DATA_FILE_PATH values.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(DATA_PATH_NAME, DATA_FILE_PATH);
+        File file = storage.getSavedData(ui);
+
+        if (file != null) {
+            tasks = new TaskList(file, ui);
+        } else {
+            tasks = new TaskList();
+        }
+    }
+
+    public String getResponse(String input) {
+        if (Parser.isDone()) {
+            System.exit(0);
+        }
+        String result = Parser.parse(input, storage, tasks, ui);
```

Comment:
Sorry just feels like result is a little vague here, are you referring to the response instead?